onboarding/tests/tests.py

```diff
@@ -0,0 +1,145 @@
+        Tests whether all the relevant user data is created
+        """
+        user = User.objects.filter(username=self.username).first()
+        email_preference = EmailPreference.objects.filter(user=user).first()
+        user_profile = UserProfile.objects.filter(user=user).first()
+        extended_profile = UserExtendedProfile.objects.filter(
+            user=user).first()
+
+        data_endpoint = settings.NODEBB_ENDPOINT + '/api/v2/users/data'
+        headers = {'Authorization': 'Bearer ' + settings.NODEBB_MASTER_TOKEN}
+        response = requests.post(data_endpoint,
+                                 data={'_uid': 1, 'username': self.username},
+                                 headers=headers
+                                 )
+
+        self.assertNotEqual(user, None)
+        self.assertNotEqual(email_preference, None)
+        self.assertNotEqual(user_profile, None)
+        self.assertNotEqual(extended_profile, None)
+        self.assertEqual(response.status_code, 200)
+
+        organization_data = Organization.objects.raw(
+            'SELECT * from onboarding_organization WHERE unclaimed_org_admin_email="{}" OR alternate_admin_email="{}"'.format(
+                self.email, self.email))
+        self.assertGreater(
+            sum(1 for row in organization_data), 0)
+
+        user_id = user.id
+        user.delete()
+
+        email_preference = EmailPreference.objects.filter(user=user).first()
+        user_profile = UserProfile.objects.filter(user=user).first()
+        extended_profile = UserExtendedProfile.objects.filter(
+            user=user).first()
+
+        self.assertEqual(email_preference, None)
+        self.assertEqual(user_profile, None)
+        self.assertEqual(extended_profile, None)
+
+        organization_data = Organization.objects.raw(
+            'SELECT * from onboarding_organization WHERE unclaimed_org_admin_email="{}" OR alternate_admin_email="{}"'.format(
+                self.email, self.email))
+        self.assertEqual(
+            sum(1 for row in organization_data), 0)
+
+        response = requests.post(data_endpoint,
+                                 data={'_uid': 1, 'username': self.username},
+                                 headers=headers
+                                 )
+        self.assertEqual(response.status_code, 400)
+
+    def tearDown(self):
+        user = User.objects.filter(username=self.username).first()
+        email_preference = EmailPreference.objects.filter(user=user).first()
+        user_profile = UserProfile.objects.filter(user=user).first()
+        extended_profile = UserExtendedProfile.objects.filter(
+            user=user).first()
+
+        if user:
+            user.delete()
+
+        if email_preference:
+            email_preference.delete()
+
+        if user_profile:
+            user_profile.delete()
+
+        if extended_profile:
+            extended_profile.delete()
+
+        NodeBBClient().users.delete_user(username=self.username)
```

Comment:
Seems we removed the usage of NodeBBClient. Do we need to remove it from here?

Reply:
removed